tasmota/xsns_47_max31865.ino

```diff
@@ -35,71 +35,137 @@
+        Pin(GPIO_SSPI_MOSI),
+        Pin(GPIO_SSPI_MISO),
+        Pin(GPIO_SSPI_SCLK)
+      );
+  }
+  if (PinUsed(GPIO_SSPI_MAX31865_CS3)) {
+      pins_used |= 1<<3;
+      max31865[3].setPins(
+        Pin(GPIO_SSPI_MAX31865_CS3),
+        Pin(GPIO_SSPI_MOSI),
+        Pin(GPIO_SSPI_MISO),
+        Pin(GPIO_SSPI_SCLK)
+      );
+  }
+  if (PinUsed(GPIO_SSPI_MAX31865_CS4)) {
+      pins_used |= 1<<4;
+      max31865[4].setPins(
+        Pin(GPIO_SSPI_MAX31865_CS4),
+        Pin(GPIO_SSPI_MOSI),
+        Pin(GPIO_SSPI_MISO),
+        Pin(GPIO_SSPI_SCLK)
+      );
+  }
+  if (PinUsed(GPIO_SSPI_MAX31865_CS5)) {
+      pins_used |= 1<<5;
+      max31865[5].setPins(
+        Pin(GPIO_SSPI_MAX31865_CS5),
+        Pin(GPIO_SSPI_MOSI),
+        Pin(GPIO_SSPI_MISO),
+        Pin(GPIO_SSPI_SCLK)
+      );
+  }
+
+  init_status=1;
+  for (uint32_t i = 0; i < MAX31865_MAX_SENSORS; i++) {
+    if (pins_used&(1<<i)) {
+        if (!max31865[i].begin(PTD_WIRES)) {
+        init_status = -1;
+      }
+    }
+  }
 }
 
 /*
 *   MAX31865_GetResult(void)
 *   Acquires the raw data via SPI, checks for MAX31865 errors and fills result structure
 */
 void MAX31865_GetResult(void){
-    uint16_t rtd;
-
-    rtd = max31865.readRTD();
-    MAX31865_Result.Rtd = rtd;
-    MAX31865_Result.PtdResistance = max31865.rtd_to_resistance(rtd, MAX31865_REF_RES);
-    MAX31865_Result.PtdTemp = max31865.rtd_to_temperature(rtd, MAX31865_PTD_RES, MAX31865_REF_RES) + MAX31865_PTD_BIAS;
+    for (uint32_t i = 0; i < MAX31865_MAX_SENSORS; i++) {
+      if (pins_used&(1<<i)){
+        uint16_t rtd;
+
+        rtd = max31865[i].readRTD();
+        MAX31865_Result[i].Rtd = rtd;
+        MAX31865_Result[i].PtdResistance = max31865[i].rtd_to_resistance(rtd, MAX31865_REF_RES);
+        MAX31865_Result[i].PtdTemp = max31865[i].rtd_to_temperature(rtd, MAX31865_PTD_RES, MAX31865_REF_RES) + MAX31865_PTD_BIAS;
+      }
+    }
 }
 
 void MAX31865_Show(bool Json){
-    char temperature[33];
-    char resistance[33];
+    for (uint32_t i = 0; i < MAX31865_MAX_SENSORS; i++) {
+      if (pins_used&(1<<i)) {
+        char temperature[33];
+        char resistance[33];
 
-    dtostrfd(MAX31865_Result.PtdResistance, Settings.flag2.temperature_resolution, resistance);
-    dtostrfd(MAX31865_Result.PtdTemp, Settings.flag2.temperature_resolution, temperature);
+        dtostrfd(MAX31865_Result[i].PtdResistance, Settings.flag2.temperature_resolution, resistance);
+        dtostrfd(MAX31865_Result[i].PtdTemp, Settings.flag2.temperature_resolution, temperature);
 
-    if(Json){
-        ResponseAppend_P(PSTR(",\"MAX31865\":{\"" D_JSON_TEMPERATURE "\":%s,\"" D_JSON_RESISTANCE "\":%s,\"" D_JSON_ERROR "\":%d}"), \
-          temperature, resistance, MAX31865_Result.ErrorCode);
+        if(Json){
+          ResponseAppend_P(PSTR(",\"MAX31865(%d)\":{\"" D_JSON_TEMPERATURE "\":%s,\"" D_JSON_RESISTANCE "\":%s,\"" D_JSON_ERROR "\":%d}"), \
+            i, temperature, resistance, MAX31865_Result[i].ErrorCode);
```

Comment:
Do not use brackets a sindex. Lok how other multisensor drivers are doing it like bh1750 allowing either a dash or underline as index.